### DISCUSSION TITLE
Adding Turso as a storage option

### DISCUSSION THREAD

**Discussion by huantrinh1802** [GothenburgBitFactory/taskwarrior#3363] [Ideas]
Since v3.*, TaskWarrior/TaskChampion starts using SQLite as a storage. Would it be feasible to add Turso as an option because it is a fork of SQLite (https://docs.turso.tech/libsql). 

With this
> SQLite Compatibility
libSQL will remain SQLite compatible by providing the same API and file format, but with additional features and optimizations.

, I would think it is do-able. 

I am happy to explore if you can help me where to look in the codebase

**Comment by djmitche** [comment]
Yes, that sounds like a cool idea!

The storage layer is implemented [here](https://github.com/GothenburgBitFactory/taskwarrior/tree/develop/taskchampion/taskchampion/src/storage) -- you'll find the trait that storage implementations must implement, as well as the sqlite mode there.

**Comment by ivishalgandhi** [comment]
i was thinking of same. just curious if libsql support was rolled out.

**Reply by djmitche** [reply]
I'm not aware of anyone having worked on this.
